utils/deliver/client_test.go

```diff
@@ -77,20 +78,21 @@ func TestBroadcastDeliver(t *testing.T) {
 
 	t.Log("One server down")
 	servers[2].Servers[0].Stop()
-	waitUntilGrpcServerIsDown(ctx, t, &servers[2].Configs[0].Endpoint)
+	listener1 := holdPort(ctx, t, servers[2].Configs[0])
```

Comment:
Earlier, we only checked whether the server was down or not. Now, we create another fake listener and explicitly close it. Can you please explain the reason as I might be missing the core issue?